ash/src/extensions/khr/video_queue.rs

```diff
@@ -0,0 +1,182 @@
+
+use crate::prelude::*;
+use crate::vk;
+use crate::RawPtr as _;
+use core::mem;
+
+impl crate::khr::video_queue::Device {
+    #[inline]
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkBindVideoSessionMemoryKHR.html>
+    pub unsafe fn bind_video_session_memory(
+        &self,
+        video_session: vk::VideoSessionKHR,
+        bind_session_memory_infos: &[vk::BindVideoSessionMemoryInfoKHR<'_>],
+    ) -> VkResult<()> {
+        (self.fp.bind_video_session_memory_khr)(
+            self.handle,
+            video_session,
+            bind_session_memory_infos.len() as u32,
+            bind_session_memory_infos.as_ptr(),
+        )
+        .result()
+    }
+
+    #[inline]
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCmdBeginVideoCodingKHR.html>
+    pub unsafe fn cmd_begin_video_coding(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        begin_info: &vk::VideoBeginCodingInfoKHR<'_>,
+    ) {
+        (self.fp.cmd_begin_video_coding_khr)(command_buffer, begin_info);
+    }
+
+    #[inline]
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCmdControlVideoCodingKHR.html>
+    pub unsafe fn cmd_control_video_coding(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        coding_control_info: &vk::VideoCodingControlInfoKHR<'_>,
+    ) {
+        (self.fp.cmd_control_video_coding_khr)(command_buffer, coding_control_info);
+    }
+
+    #[inline]
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCmdEndVideoCodingKHR.html>
+    pub unsafe fn cmd_end_video_coding(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        end_coding_info: &vk::VideoEndCodingInfoKHR<'_>,
+    ) {
+        (self.fp.cmd_end_video_coding_khr)(command_buffer, end_coding_info);
+    }
+
+    #[inline]
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCreateVideoSessionKHR.html>
+    pub unsafe fn create_video_session(
+        &self,
+        create_info: &vk::VideoSessionCreateInfoKHR<'_>,
+        allocation_callbacks: Option<&vk::AllocationCallbacks<'_>>,
+    ) -> VkResult<vk::VideoSessionKHR> {
+        let mut video_session = mem::zeroed();
+        (self.fp.create_video_session_khr)(
+            self.handle,
+            create_info,
+            allocation_callbacks.as_raw_ptr(),
+            &mut video_session,
+        )
+        .result_with_success(video_session)
+    }
+
+    #[inline]
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCreateVideoSessionParametersKHR.html>
+    pub unsafe fn create_video_session_parameters(
+        &self,
+        create_info: &vk::VideoSessionParametersCreateInfoKHR<'_>,
+        allocation_callbacks: Option<&vk::AllocationCallbacks<'_>>,
+    ) -> VkResult<vk::VideoSessionParametersKHR> {
+        let mut video_session_parameters = mem::zeroed();
```

Comment:
again `MaybeUninit`. Don't forget the `assume_init_on_success()` helper in place of `.result_with_success()`.